From cae5c1e904b72f20e94864ca048f878897f724e8 Mon Sep 17 00:00:00 2001
From: Zataron <you@example.com>
Date: Tue, 24 May 2011 16:05:20 +0200
Subject: [PATCH] Introducing a method to fix bounding radius/combat reach of mobs

---
 src/game/Object.h |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/src/game/Object.h b/src/game/Object.h
index 05517c1..c16e3ff 100644
--- a/src/game/Object.h
+++ b/src/game/Object.h
@@ -190,6 +190,14 @@ class MANGOS_DLL_SPEC Object
             return m_floatValues[ index ];
         }
 
+        void SetBoundingValue(uint8 t, float v)
+        {
+            if (t == 0)
+                m_floatValues[UNIT_FIELD_COMBATREACH] = v;
+            else if (t == 1)
+                m_floatValues[UNIT_FIELD_BOUNDINGRADIUS] = v;
+        }
+
         uint8 GetByteValue( uint16 index, uint8 offset) const
         {
             MANGOS_ASSERT( index < m_valuesCount || PrintIndexError( index , false) );
-- 
1.7.4.msysgit.0

