From 77d9418bb91244ef31e47c054be767f0674d175b Mon Sep 17 00:00:00 2001
From: Zataron <you@example.com>
Date: Sat, 22 Jan 2011 17:51:33 +0100
Subject: [PATCH] Crit and dodge formulas

---
 src/game/Player.cpp |   15 ++++++++++-----
 1 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/src/game/Player.cpp b/src/game/Player.cpp
index 563b821..36dcc78 100644
--- a/src/game/Player.cpp
+++ b/src/game/Player.cpp
@@ -4602,8 +4602,9 @@ uint32 Player::GetShieldBlockValue() const
 
 float Player::GetMeleeCritFromAgility()
 {
-  // from mangos 3462 for 1.12
-  float val=0.0f,classrate = 0.0f;
+    // from mangos 3462 for 1.12
+    float val = 0.0f, classrate = 0.0f, LevelFactor = 0.0f, fg = 0.0f;
+
     // critical
     switch(getClass())
     {
@@ -4619,24 +4620,30 @@ float Player::GetMeleeCritFromAgility()
         default:            classrate = 20.0f; break;
     }
 
-    val = GetStat(STAT_AGILITY)/classrate;
-  return val;
+    fg = (0.35f*(float)(getLevel())) + 5.55f;
+    LevelFactor = 106.20f / fg - 3;
+    //LevelFactor = 1 for lvl 60 chars
+    val = LevelFactor * (GetStat(STAT_AGILITY)/classrate);
+
+    return val;
 }
 
 float Player::GetDodgeFromAgility()
 {
-     // from mangos 3462 for 1.12
-    float val=0,classrate=0;
+    // from mangos 3462 for 1.12
+    float val=0,classrate=0,LevelRate=0;
 
     //dodge
+    //LevelRate = 1 for lvl 60 chars
+    LevelRate = ((16.225f/((0.45f*(float)(getLevel()))+2.5f))-0.1f)/0.45f;
     if(getClass() == CLASS_HUNTER) classrate = 26.5;
     else if(getClass() == CLASS_ROGUE)  classrate = 14.5;
     else classrate = 20;
     ///*+(Defense*0,04);
     if (getRace() == RACE_NIGHTELF)
-        val = GetStat(STAT_AGILITY)/classrate + 1;
+        val = LevelRate * (GetStat(STAT_AGILITY)/classrate) + 1;
     else
-        val = GetStat(STAT_AGILITY)/classrate;
+        val = LevelRate * (GetStat(STAT_AGILITY)/classrate);
 
     return val;
 
