From 09a7ab5a374284929e4b673d40c975b5bddc0026 Mon Sep 17 00:00:00 2001
From: Craig <hippouss@.(none)>
Date: Tue, 23 Mar 2010 18:00:32 +0100
Subject: [PATCH 2/6] Fall under map ressurect

---
 src/game/MovementHandler.cpp |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/game/MovementHandler.cpp b/src/game/MovementHandler.cpp
index 81f8bd0..c78ba4d 100644
--- a/src/game/MovementHandler.cpp
+++ b/src/game/MovementHandler.cpp
@@ -351,6 +351,9 @@ void WorldSession::HandleMovementOpcodes( WorldPacket & recv_data )
 
                 // cancel the death timer here if started
                 plMover->RepopAtGraveyard();
+
+                plMover->ResurrectPlayer(0.5f);
+                plMover->SpawnCorpseBones();
             }
         }
     }
-- 
1.6.5.1.1367.gcd48

