From 4d294bce00097bd57ec4534105502f271558b068 Mon Sep 17 00:00:00 2001
From: Craig <hippouss@.(none)>
Date: Sun, 6 Mar 2011 15:39:44 +0100
Subject: [PATCH] vmaps

---
 src/game/vmap/VMapDefinitions.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/game/vmap/VMapDefinitions.h b/src/game/vmap/VMapDefinitions.h
index 9c496f5..d4af735 100644
--- a/src/game/vmap/VMapDefinitions.h
+++ b/src/game/vmap/VMapDefinitions.h
@@ -25,8 +25,8 @@
 
 namespace VMAP
 {
-    const char VMAP_MAGIC[] = "VMAPz3.0";                   // used in final vmap files
-    const char RAW_VMAP_MAGIC[] = "VMAPz03";                // used in extracted vmap files with raw data
+    const char VMAP_MAGIC[] = "VMAPs3.0";                   // used in final vmap files
+    const char RAW_VMAP_MAGIC[] = "VMAPs03";                // used in extracted vmap files with raw data
 
     // defined in TileAssembler.cpp currently...
     bool readChunk(FILE *rf, char *dest, const char *compare, uint32 len);
-- 
1.7.3.1.msysgit.0

